tensorboard/plugins/mesh/polymer3/tf_mesh_dashboard/tf-mesh-dashboard.ts

```diff
@@ -209,123 +205,123 @@ <h3>No point cloud data was found.</h3>
-              // No need to update anything if there are no changes.
-              return;
-            }
-            const runToTag = _.mapValues(runToTagInfo, (x) => Object.keys(x));
-            const tags = tf_backend.getTags(runToTag);
-            this.set('_dataNotFound', tags.length === 0);
-            this.set('_runToTagInfo', runToTagInfo);
-          });
-        },
-        _reloadMeshes() {
-          this._getAllChildren().forEach((g) => {
-            g.reload();
-          });
-        },
-        _makeCategories(runToTagInfo, selectedRuns, tagFilter) {
-          const runToTag = _.mapValues(runToTagInfo, (x) => Object.keys(x));
-          const baseCategories = tf_categorization_utils.categorizeRunTagCombinations(
-            runToTag,
-            selectedRuns,
-            tagFilter
-          );
-
-          function explodeItem(item) {
-            const samples = runToTagInfo[item.run][item.tag].samples;
-            return _.range(samples).map((i) =>
-              Object.assign({}, item, {
-                sample: i,
-                ofSamples: samples,
-              })
-            );
-          }
-          const withSamples = baseCategories.map((category) =>
-            Object.assign({}, category, {
-              items: [].concat.apply([], category.items.map(explodeItem)),
-            })
-          );
-          return withSamples;
-        },
-      }); // End of Polymer constructor call.
-    })(); // End of anonymous namespace.
-  </script>
-</dom-module>
+  `;
+
+  @property({type: Boolean})
+  reloadOnReady: boolean = true;
+
+  @property({type: Array})
+  _selectedRuns: string[];
+
+  @property({type: Object})
+  _runToTagInfo: object;
+
+  @property({type: Boolean})
+  _dataNotFound: boolean;
+
+  @property({type: String})
+  _tagFilter: string = '.*';
+
+  @property({
+    type: String,
+    notify: true,
+  })
+  _selectedView: string = 'all';
+
+  @property({type: Object})
+  _requestManager = new RequestManager();
+
+  constructor() {
+    super();
+    window.addEventListener(
+      'resize',
+      () => {
+        this._handleWindowResize();
+      },
+      false
+    );
+    if (this.reloadOnReady) this.reload();
+  }
+
+  _getAllChildren() {
+    return Array.from(
+      this.shadowRoot.querySelectorAll('tf-mesh-loader')
+    ) as TfMeshLoader[];
+  }
+
+  _onCameraPositionChanged(event) {
+    if (this._selectedView == 'share') {
+      this._getAllChildren().forEach((g) => {
+        if (event.target == g) return; // Do not update trigger camera.
+        (g as any).setCameraViewpoint(
+          event.detail.position,
+          event.detail.far,
+          event.detail.target
+        );
+      });
+    }
+  }
+
+  _shouldOpen(index) {
+    return index <= 2;
+  }
+
+  reload() {
+    this._fetchTags().then(this._reloadMeshes.bind(this));
+  }
+
+  _handleWindowResize() {
+    this._getAllChildren().forEach((g) => {
+      (g as any).redraw();
```

Comment:
Can we add `redraw()` to the `TfMeshLoader` public interface, so that this doesn't require `as any`?  It looks like `redraw()` is meant to be public anyway.

Reply:
Done.